GaelAlburo/final-project
# Feat(users)/adding context

**Summary**
This PR introduces context management for user data, local storage integration, and API functionality for user operations. These changes enhance state management and persistence across the application.

**Changes**
Create Users API (feature/Create_users_API):
  Implemented a users API to handle CRUD operations for user data.
  Includes error handling and loading states for API requests.

Local Storage Integration (feature/adding_LocalStorage):
  Added functionality to persist user data in localStorage.
  Ensures user sessions persist even after a page refresh.

User Context (feature/adding_context):
  Created a context provider for user data using React's Context API.
  Allows centralized management of user state and ensures consistent access across components.

**How to Test**
Run the application locally (npm run dev).
Test the user-related API endpoints to ensure data retrieval and manipulation work correctly.
Verify that user sessions persist across page reloads.
Check that the context provides accurate and consistent user state to components.

**Notes**
The context integration ensure seamless user experience.
